### PR TITLE
Guido/ab5020 auto publish schema

### PR DESCRIPTION
**Pull request by gkoller**
This is a fairly large change. De individual commits have been extensively documented, so have a look at them too. In short this PR does the following:

- change project structure to allow for easy packaging into a proper Python package
- Python package includes all schemas and exposes a `publish` command line command.
- Simplified `Dockerfile` that simply installs the Python package
- `Jenkinsfile` that:
  - builds Docker images
  - for both environments: `acceptance` and `production` (the latter **only after approval in Jenkins**)
    - tag image
    - publish schemas
    - import schemas into DSO-API DB. 

The Jenkins job that utilizes the new functionality is: https://ci.data.amsterdam.nl/job/DataServices/job/auto_publish_ams_schemas/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,34 +1 @@
...

    stage("Import schemas into DSO-API PROD DB") {
        tryStep "import", {
            build job: 'ams_schema_importer',
                parameters: [
                    [$class: 'StringParameterValue', name: 'environment', value: 'production']
                ]
        }
    }
}
```
Comment (ngaranko) [inline Jenkinsfile:1]: `^M` not cool.

File: src/amsterdam_schema/publish/cli.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
+__doc__ = """
+
+Script that publishes verified schemas to the objectstore.
+
+Course of action is as follows:
+
+* A zipfile of the master branch of the 'amsterdam-schema' repo is fetched from github
```
Comment (ngaranko) [inline src/amsterdam_schema/publish/cli.py:7]: This is not happening anymore.

**Review by jjmurre — APPROVED** [review]
Look good! Now nicely packaged. And, the local publishing still works!

File: Jenkinsfile (hunk 1 above)
Comment (gkoller) [reply Jenkinsfile:1]: That's bizarre. I did notice the pre-commit _Trim Trailing Whitespace_ running twice in succession on this file, but can't imagine it doing something like that. Anyway should be fixed in my next push.

File: src/amsterdam_schema/publish/cli.py (hunk 2 above)
Comment (gkoller) [reply src/amsterdam_schema/publish/cli.py]: Good catch.

File: Jenkinsfile [hunk 3]
```diff
@@ -17,18 +17,83 @@ def tryStep(String message, Closure block, Closure tearDown = null) {
...
+                    parameters: [
+                            [$class: 'StringParameterValue', name: 'INVENTORY', value: 'acceptance'],
+                            [$class: 'StringParameterValue', name: 'PLAYBOOK', value: 'publish-dataservices-static-schemas.yml'],
+                    ]
+       }
+   }
+
+    stage("Import schemas into DSO-API ACC DB") {
+        tryStep "import", {
+            build job: 'ams_schema_importer',
```
Comment (cvriel) [inline Jenkinsfile:57]: Mooi dat je deze stap hebt kunnen integreren, dat scheelt weer een extra handeling

File: README.md [hunk 4]
```diff
@@ -1,12 +1,18 @@
 <div align="center">
-  <img width="100px" src="amsterdam-schema.svg" />
+  <img alt="Amsterdam logo" width="100px" src="amsterdam-schema.svg" />
 </div>
 
 # Amsterdam Schema
 
-Amsterdam Schema aims to describe and validate [open data published by the City of Amsterdam](https://api.data.amsterdam.nl/api/), in order to make the storing and publishing of different datasets more structured, simpler and better documented.
+Amsterdam Schema aims to describe and validate
+[open data published by the City of Amsterdam](https://api.data.amsterdam.nl/api/),
```
Comment (cvriel) [inline README.md:8]: Je hebt deze laten staan omdat we op termijn de catalogus op deze url ook toegankelijk haan maken https://api.data.amsterdam.nl/api/?

File: README.md [hunk 5]
```diff
@@ -70,3 +98,61 @@ For more information, see (some of these pages are in Dutch):
...
+```
+
+but it likely isn't.
+
+See, the `publish` tool expects a number of environment variables to be set.
+These are:
+
+```console
+DATAPUNT_ENVIRONMENT=...
+OS_USERNAME="dataservices
```
Comment (cvriel) [inline README.md:128]: Quote te veel?
`OS_USERNAME="dataservices`

File: README.md [hunk 6]
```diff
@@ -70,3 +98,61 @@ For more information, see (some of these pages are in Dutch):
...
+% publish
+```
+
+but it likely isn't.
+
+See, the `publish` tool expects a number of environment variables to be set.
+These are:
+
+```console
+DATAPUNT_ENVIRONMENT=...
```
Comment (cvriel) [inline README.md:127]: Aangeven wat we hier kunnen invullen o.i.d.?
DATAPUNT_ENVIRONMENT=[acceptation|production]

**Review by cvriel — APPROVED** [review]
De CLI ziet er compact en wmbt goed uit

**Review by ngaranko — APPROVED** [review]
LGTM.

File: README.md (hunk 4 above)
Comment (gkoller) [reply README.md:8]: Actually this appeared in the diff due to some reformatting work (https://rhodesmill.org/brandon/2012/one-sentence-per-line/) in prep op adding new documentation. I have no idea if this part is still relevant or what in entails for the future.

File: README.md (hunk 5 above)
Comment (gkoller) [reply README.md:128]: Will fix

File: README.md (hunk 6 above)
Comment (gkoller) [reply README.md:127]: Will do.Thanks!

File: README.md (hunk 4 above)
Comment (gkoller) [reply README.md:8]: Leaving it in for now. Not really part of the PR.

**Pull request merged**